pinot-core/src/main/java/org/apache/pinot/core/data/readers/PinotSegmentRecordReader.java

```diff
@@ -166,6 +192,6 @@ public void close()
     for (PinotSegmentColumnReader columnReader : _columnReaderMap.values()) {
       columnReader.close();
     }
-    _immutableSegment.destroy();
+    _indexSegment.destroy();
```

Comment:
We don't want to destroy the segment if it is passed in instead of loaded in this class